src/main/java/slimeknights/tconstruct/plugin/jei/casting/AbstractCastingCategory.java

```diff
@@ -0,0 +1,86 @@
+package slimeknights.tconstruct.plugin.jei.casting;
+
+import mezz.jei.api.constants.VanillaTypes;
+import mezz.jei.api.gui.IRecipeLayout;
+import mezz.jei.api.gui.drawable.IDrawable;
+import mezz.jei.api.gui.drawable.IDrawableAnimated;
+import mezz.jei.api.gui.ingredient.IGuiFluidStackGroup;
+import mezz.jei.api.gui.ingredient.IGuiItemStackGroup;
+import mezz.jei.api.helpers.IGuiHelper;
+import mezz.jei.api.ingredients.IIngredients;
+import mezz.jei.api.recipe.category.IRecipeCategory;
+import net.minecraft.block.Block;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.fluids.FluidStack;
+import slimeknights.tconstruct.library.Util;
+import slimeknights.tconstruct.smeltery.recipe.AbstractCastingRecipe;
+
+public abstract class AbstractCastingCategory<T extends AbstractCastingRecipe> implements IRecipeCategory<T> {
```

Comment:
Is this still WIP? I see a lot of commented code and the cooling time does not seem done.

Reply:
Having a bit of fun figuring out how to make the arrow do the same as cooling time. Perhaps not
such a good idea, as some things cool in 2s, which would look pretty spastic.

Reply:
Mezz just went for 1/20th of the time in JEI IIRC. Look at the furnace recipe.

Reply:
Problem is all their furnace/etc recipe stuff has a fixed time. we don't. They pass time in
via the ctor.

Reply:
Oh right, forgot they did that. There might be a way to tell it a time when you draw it, but its not really a high priority.